mkdocs/config/config_options.py

```diff
@@ -468,6 +468,9 @@ def post_validation(self, config, key_name):
             raise ValidationError("The path set in {name}.custom_dir ('{path}') does not exist.".
                                   format(path=theme_config['custom_dir'], name=key_name))
 
+        if 'locale' in theme_config and not isinstance(theme_config['locale'], str):
+            raise ValidationError("'{name}.locale' must be a string.".format(name=theme_config['name']))
```

Comment:
I'm thinking perhaps we should validate that `theme.locale ` contains a known valid language code (not just that it is a string). That should be easy enough to do as `babel` includes the code for that. Presumably, we would only run the validation test if babel is installed.

It might also be helpful to validate that a translation exists for the given language code. However, that would presumably require checking that the necessary translation files exist. Does babel have any existing code for that?

Any thoughts of either validation proposal?

Reply:
On second thought, maybe we don't need to. After all, in `mkdocs.localization.install_translations` we call `babel.core.Local.parse`. Presumably that would raise any validation errors. Although, it might make sense to catch them and reraise as a Mkdocs `ValidationError`. And this won't catch attempts to use a locale with no translation defined. I know we document that that will fallback to English, but users might expect at least a warning about it.